src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -93,40 +99,55 @@ public Class<? extends Entity> getReturnType() {
 	}
 	
 	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
+	public String toString(@Nullable Event e, boolean debug) {
 		if (e == null)
 			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
 		return Classes.getDebugMessage(getAll(e));
 	}
 	
 	@Override
-	public boolean setTime(final int time) {
+	public boolean setTime(int time) {
 		return super.setTime(time, EntityTargetEvent.class, getExpr());
 	}
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
-			return CollectionUtils.array(LivingEntity.class);
-		return super.acceptChange(mode);
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case RESET:
+			case DELETE:
+			case REMOVE:
+				return CollectionUtils.array(LivingEntity.class);
+		}
+		return null;
 	}
 	
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
-			final LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
-			for (final LivingEntity entity : getExpr().getArray(e)) {
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.RESET) {
+			LivingEntity target = (delta == null || mode == ChangeMode.RESET) ? null : (LivingEntity) delta[0]; // null will make the entity target-less (reset target)
+
+			for (LivingEntity entity : getExpr().getArray(e)) {
 				if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
 					((EntityTargetEvent) e).setTarget(target);
 				} else {
 					if (entity instanceof Creature)
 						((Creature) entity).setTarget(target);
 				}
 			}
+		} else if (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE) {
+			for (LivingEntity en : getExpr().getArray(e)) {
+				removeTarget(Utils.getTarget(en, type));
+			}
 		} else {
 			super.change(e, delta, mode);
 		}
 	}
+
+	private void removeTarget(@Nullable Entity e) {
+		if (e != null && !(e instanceof OfflinePlayer))
+			e.remove();
```

Comment:
I'm not sure if this is what we want to do. I guess you could interpret `remove target of entity` in multiple ways:
- delete the target of the entity from the world
- act the same as RESET

IMO I think it's better if we just treat DELETE like RESET (also because `clear target of entity` seems to align better with that)

Reply:
For me and for many DELETE was the way to remove entities from world without killing them and making kill sounds/dropping items, this is useful in minigames/magic tricks/etc.

Reply:
I think it's fine that delete has different interpretations depending on the input, because whole expression is like this, for player it returns what they're looking at, for entities the actual target.

Reply:
Also, I don't think we need a method for these 2 lines